include/hip/hcc_detail/math_functions.h

```diff
@@ -22,6 +22,7 @@ THE SOFTWARE.
 
 #pragma once
 
+#include "hip_fp16_math_fwd.h"
```

Comment:
This causes build failures in the ```directed_tests/deviceLib/hipTestIncludeMath```. To fix this should either be included after ```hip/hcc_details/host_defines.h``` or we should modify ```hip_fp16_math_fwd.h``` to include ```host_defines.h```.